src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsVersionCheck.cs

```diff
@@ -47,9 +48,10 @@ public async Task<UpgradeReadiness> IsReadyAsync(IProject project, UpgradeReadin
             var components = await project.GetComponentsAsync(token).ConfigureAwait(false);
             if (components.HasFlag(ProjectComponents.XamarinAndroid) || components.HasFlag(ProjectComponents.XamariniOS))
             {
-                if (!DoesHaveMinimumXamarinVersion(project, _comparer))
+                if (!DoesHaveMinimumXamarinVersion(project, _comparer)
+                    && project.NuGetReferences.PackageReferences.Any(x => x.Name.Equals("Xamarin.Forms", StringComparison.OrdinalIgnoreCase)))
                 {
-                    _logger.LogError(XAMARIN_FORMS_MESSAGE, CATEGORY, UPGRADE_LINK);
+                    _logger.LogError(UpgradeMessage);
```

Comment:
This shouldn't have been a problem

Reply:
To answer both the above question and this one:  LogError wasn't the problem but during the [CurrentProjectSelectionStep](https://github.com/dotnet/upgrade-assistant/blob/5d07f3b86b233108d705e3c0549ca845e5e54964/src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Solution/CurrentProjectSelectionStep.cs#L227), the failure occurs when the UpgradeGuidanceMessages are adding the UpgradeMessage itself. SR.Format fails [here](https://github.com/dotnet/upgrade-assistant/blob/df767234239084ae0461612b2aeb7f459e3ab6ac/src/extensions/maui/Microsoft.DotNet.UpgradeAssistant.Extensions.Maui/XamarinFormsVersionCheck.cs#L33).

Reply:
The string interpolation fixes the issue in the way that there is no formatting error in SR.Format but it also doesn't do what it's supposed to do.  Why don't we keep UpgradeMessage as the value for the LogError to use the same value in both areas and update XAMARIN_FORMS_MESSAGE to setup UpgradeMessage correctly. SR.Format works as intended with the new changes.